python/paddle/fluid/dygraph/layers.py

```diff
@@ -752,6 +752,40 @@ def __delattr__(self, name):
         else:
             object.__delattr__(self, name)
 
+    def __dir__(self):
+        """
+        Get all parameters, buffers(non-parameter variables), sublayers, method and attr of Layer.
+
+        Examples:
+            import paddle.fluid as fluid
+            import numpy as np
+
+            fluid.dygraph.enable_dygraph()
+
+            class Mylayer(fluid.dygraph.Layer):
+                def __init__(self):
+                    super(Mylayer1, self).__init__()
+                    self.linear1 = fluid.dygraph.Linear(10, 10)
+                    self.linear2 = fluid.dygraph.Linear(5, 5)
+                    self.conv2d = fluid.dygraph.Conv2D(3, 2, 3)
+                    self.embedding = fluid.dygraph.Embedding(size=[128, 16])
+                    self.h_0 = fluid.dygraph.to_variable(np.zeros([10, 10]).astype('float32'))
+
+            mylayer = Mylayer()
+            print(dir(mylayer))
```

Comment:
Better add the output of the example code

Reply:
Ok

Reply:
Done